explorer/src/components/account/nftoken-hooks.tsx

```diff
@@ -0,0 +1,39 @@
+import { NftokenFetcher, NftokenTypes } from "@glow-xyz/nftoken-js";
+import useSWR, { SWRResponse } from "swr";
+import { useCluster } from "../../providers/cluster";
+
+export const useCollectionNfts = ({
+  collectionAddress,
+}: {
+  collectionAddress: string;
+}): {
+  // We can be confident that data will be nonnull even if the request fails,
+  // if we defined fallbackData in the config.
+  data?: NftokenTypes.NftInfo[];
+  error: any;
+  mutate: SWRResponse<NftokenTypes.NftInfo[], never>["mutate"];
+} => {
+  const { url } = useCluster();
+  const swrKey = [collectionAddress, "getNftsInCollection"];
+  const { data, error, mutate } = useSWR(swrKey, async () => {
+    return await NftokenFetcher.getNftsInCollection({
+      collection: collectionAddress,
+      rpcUrl: url,
+    });
+  });
```

Comment:
Your fetcher must be stable (same function every render) and can't close over other variables in scope. All the information it needs, you can pass through arguments!

```typescript
async function nftsInCollectionFetcher(([_, collectionAddress, url]) => {
  return await NftokenFetcher.getNftsInCollection({
    collection: collectionAddress,
    rpcUrl: url,
  });
});

export const useCollectionNfts(...) {
  /* ... */
  const {data, error, mutate} = useSWR(
    ['getNftsInCollection, collectionAddress, url],
    nftsInCollectionFetcher,
  );
  /* ... */
}
```

This way:

* SWR won't blow your result away every render because you passed a new fetcher
* Changes to `url` will _actually_ result in data being stored in your SWR cache with that URL as a key

Reply:
Hmm, I'm not sure you are correct that SWR will blow away the result because you passed a new fetcher. What makes you think that?

Fetcher is stored as a `ref`:

https://github.com/vercel/swr/blob/main/core/use-swr.ts#L90

Reply:
I don't think this is an issue but made it stable anyways